community/values/src/main/java/org/neo4j/values/storable/ValueComparator.java

```diff
@@ -67,13 +68,13 @@ public Integer ternaryCompare( Value v1, Value v2 )
         {
             return v1.unsafeTernaryCompareTo( v2 );
         }
-        return x;
+        return Comparison.from( x );
```

Comment:
This is wrong, but that wrong was not introduced by this PR. If the value-groups are not the same, `ternaryCompare` should give `Comparison.UNDEFINED`. We need to go over this logic later.